### PR TITLE
hyprland-monitor-attached: init at 0.1.5

### PR DESCRIPTION
**Pull request by bddvlpr**
## Description of changes

Requested at #282397, tested and working. Doesn't have actual tagging / version releases, based on last commit at 2023-10-03.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by bddvlpr** [comment]
Result of `nixpkgs-review pr 283262` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hyprland-monitor-attached</li>
  </ul>
</details>

File: pkgs/by-name/hy/hyprland-monitor-attached/package.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ rustPlatform, fetchFromGitHub, lib }:
+
+rustPlatform.buildRustPackage rec {
```
[the remark this answers is not in the document]
Comment (bddvlpr) [reply pkgs/by-name/hy/hyprland-monitor-attached/package.nix:3]: Afaik, recursive attributes are only a thing in mkDerivation and some build-support functions, or am I missing something?

[the remark this answers is not in the document]
Comment (jtbx) [reply pkgs/by-name/hy/hyprland-monitor-attached/package.nix:3]: Ah that's right, my bad.

**Review by jtbx — APPROVED** [review]

File: pkgs/by-name/hy/hyprland-monitor-attached/package.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "hyprland-monitor-attached";
+  version = "unstable-2023-10-03";
+
+  src = fetchFromGitHub {
+    owner = "coffebar";
+    repo = pname;
```
Comment (donovanglover) [inline pkgs/by-name/hy/hyprland-monitor-attached/package.nix:9]:
```suggestion
    repo = "hyprland-monitor-attached";
```
Avoid `repo = pname;`

File: pkgs/by-name/hy/hyprland-monitor-attached/package.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ rustPlatform, fetchFromGitHub, lib }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "hyprland-monitor-attached";
+  version = "unstable-2023-10-03";
```
Comment (donovanglover) [inline pkgs/by-name/hy/hyprland-monitor-attached/package.nix:5]:
```suggestion
  version = "0.1.5-unstable-2023-10-03";
```

Version is updated in [`Cargo.toml`](https://github.com/coffebar/hyprland-monitor-attached/commits/main/Cargo.toml)

**Review by donovanglover — COMMENTED** [review]
Remember to update the version in the commit message

File: pkgs/by-name/hy/hyprland-monitor-attached/package.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner = "coffebar";
+    repo = pname;
+    rev = "72551d65bfec5afa08f4053620415e6e5f64d0e1";
```
Comment (donovanglover) [inline pkgs/by-name/hy/hyprland-monitor-attached/package.nix:10]:
```suggestion
    rev = version;
```
Upstream tagged the latest version

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 283262` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hyprland-monitor-attached</li>
  </ul>
</details>

**Review by donovanglover — APPROVED** [review]
LGTM

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**